### ISSUE TITLE
Record parameters of generic types should give safety warnings on default values

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#47344]
**Version Used**: e7bad77

**Steps to Reproduce**:

```cs
#nullable enable

var rec = new Rec1<string>();
rec.t.ToString(); // NRE at runtime

record Rec1<T>(T t = default) // no warning
{
}
```

**Expected Behavior**: A safety warning is given on the implicit assignment to the synthesized property `Rec1<T>.t`.

**Actual Behavior**: No warnings are given.

**Remarks**:

Once we implement non-safety warnings for unconstrained generics in C# 9 we will be halfway there, but to handle this scenario correctly we need to figure out how to distinguish compiler-generated nodes that we care about (e.g. record synthesized constructors) from compiler-generated nodes that we don't care about (e.g. synthesized default arguments at call sites).

If we just start warning on all compiler generated assignments, we will also get an undesirable warning at the `new Rec1` creation site, since we synthesize an argument for the optional parameter for flow analysis purposes.

**Comment by RikkiGibson** [comment]
I expect to be able to solve this at the same time as #40818
